Forward-port changes from nv-procfs.c in 361.18 in order to build against newer kernels.
--- nvidia-drivers-355.11-r99/work/kernel/nvidia/nv-procfs.c	2015-08-27 00:23:15.000000000 +0200
+++ nvidia-drivers-361.18-r99/work/kernel/nvidia/nv-procfs.c	2016-01-10 05:03:42.000000000 +0100
@@ -40,8 +40,6 @@
 static struct proc_dir_entry *proc_nvidia_patches;
 static struct proc_dir_entry *proc_nvidia_gpus;
 
-extern nv_linux_state_t *nv_linux_devices;
-
 extern char *NVreg_RegistryDwords;
 extern char *NVreg_RmMsg;
 extern char *NVreg_AssignGpus;
@@ -360,7 +358,8 @@
     registry_keys = ((nvl != NULL) ?
             nvl->registry_keys : nv_registry_keys);
 
-    return seq_printf(s, "Binary: \"%s\"\n", registry_keys);
+    seq_printf(s, "Binary: \"%s\"\n", registry_keys);
+    return 0;
 }
 
 static ssize_t
@@ -560,7 +559,8 @@
     void *v
 )
 {
-    return seq_puts(s, s->private);
+    seq_puts(s, s->private);
+    return 0;
 }
 
 NV_DEFINE_PROCFS_SINGLE_FILE(text_file);
